pkg/chains/formats/slsa/internal/resolved_dependencies/resolved_dependencies.go

```diff
@@ -86,3 +99,198 @@ func RemoveDuplicateResolvedDependencies(resolvedDependencies []v1.ResourceDescr
+		return &slsa.ResourceDescriptor{
+			Name:   PipelineTaskConfigName,
+			URI:    tr.Status.Provenance.RefSource.URI,
+			Digest: tr.Status.Provenance.RefSource.Digest,
+		}, nil
+	}
+	return nil, nil
+}
+
+// fromPipelineTask adds the resolved dependencies from pipeline tasks
+// such as pipeline task uri/digest for remote pipeline tasks and step and sidecar images.
+func fromPipelineTask(logger *zap.SugaredLogger, pro *objects.PipelineRunObjectV1, addTasks addTaskDescriptorContent) ([]slsa.ResourceDescriptor, error) {
+	pSpec := pro.Status.PipelineSpec
+	resolvedDependencies := []slsa.ResourceDescriptor{}
+	if pSpec != nil {
+		pipelineTasks := append(pSpec.Tasks, pSpec.Finally...)
+		for _, t := range pipelineTasks {
+			tr := pro.GetTaskRunFromTask(t.Name)
+			// Ignore Tasks that did not execute during the PipelineRun.
+			if tr == nil || tr.Status.CompletionTime == nil {
+				logger.Infof("taskrun status not found for task %s", t.Name)
+				continue
+			}
+			rd, err := addTasks(tr)
+			if err != nil {
+				logger.Errorf("error storing taskRun %s, error: %s", t.Name, err)
+				continue
+			}
+
+			if rd != nil {
+				resolvedDependencies = append(resolvedDependencies, *rd)
+			}
+
+			mats := []common.ProvenanceMaterial{}
+
+			// add step images
+			stepMaterials, err := material.FromStepImages(tr)
+			if err != nil {
+				return nil, err
+			}
+			mats = append(mats, stepMaterials...)
+
+			// add sidecar images
+			sidecarMaterials, err := material.FromSidecarImages(tr)
+			if err != nil {
+				return nil, err
+			}
+			mats = append(mats, sidecarMaterials...)
+
+			// convert materials to resolved dependencies
+			resolvedDependencies = append(resolvedDependencies, ConvertMaterialsToResolvedDependencies(mats, "")...)
+		}
+	}
+	return resolvedDependencies, nil
+}
+
+// taskDependencies gather all dependencies in a task and adds them to resolvedDependencies
+func taskDependencies(ctx context.Context, tro *objects.TaskRunObjectV1) ([]slsa.ResourceDescriptor, error) {
+	var resolvedDependencies []slsa.ResourceDescriptor
+	var err error
+	mats := []common.ProvenanceMaterial{}
+
+	// add step and sidecar images
+	stepMaterials, err := material.FromStepImages(tro)
+	mats = append(mats, stepMaterials...)
+	if err != nil {
+		return nil, err
+	}
+	sidecarMaterials, err := material.FromSidecarImages(tro)
+	if err != nil {
+		return nil, err
+	}
+	mats = append(mats, sidecarMaterials...)
+	resolvedDependencies = append(resolvedDependencies, ConvertMaterialsToResolvedDependencies(mats, "")...)
+
+	mats = material.FromTaskParamsAndResults(ctx, tro)
+	// convert materials to resolved dependencies
+	resolvedDependencies = append(resolvedDependencies, ConvertMaterialsToResolvedDependencies(mats, InputResultName)...)
+
+	// add task resources
+	// =====
+	// convert to v1beta1 and add any task resources
+	serializedResources := tro.Annotations["tekton.dev/v1beta1-spec-resources"]
```

Comment:
Can we move the annotation name to a variable?

Do we need to only perform the steps below if `serializedResources` is not empty?

Reply:
Sure! annotation moved to a variable.

If `serializedResources` is empty the `json.Marshall` call will fail, and won't set the `shouldReplace` to `true`. We still have to try to get the results from v1beta1 TaskRun object, but we are reading the taskrun resources from another source. So I think it make sense to not check if `serializedResources` is empty.